### PR TITLE
[Offline] if order details are set don't set it again

### PR DESCRIPTION
**Pull request by antonioperic**

### CODE REVIEW

File: src/Payum/Offline/Action/FillOrderDetailsAction.php [hunk 1]
```diff
@@ -16,6 +16,10 @@ class FillOrderDetailsAction implements ActionInterface
     public function execute($request)
     {
         RequestNotSupportedException::assertSupports($this, $request);
+        
+        if($request->getToken()->getDetails()) {
+            return;
+        }
```
Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: the values from order should be forced and overwrite values in details, custom ones could be set like this

https://github.com/Payum/Offline/pull/2/files#r24068718

Comment (antonioperic) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: So you want to have default for paid and then if exist in order to overide it?

Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: yeap, I think setting it to true is more spread use case

**Comment by antonioperic** [comment]
Like this?

File: src/Payum/Offline/Action/FillOrderDetailsAction.php [hunk 2]
```diff
@@ -16,8 +16,15 @@ class FillOrderDetailsAction implements ActionInterface
     public function execute($request)
     {
         RequestNotSupportedException::assertSupports($this, $request);
+        
+        if($request->getToken()->getDetails()) {
```
Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: what is this for?

File: src/Payum/Offline/Action/FillOrderDetailsAction.php [hunk 3]
```diff
@@ -26,7 +33,7 @@ public function execute($request)
         $details['description'] = $order->getDescription();
         $details['client_email'] = $order->getClientEmail();
         $details['client_id'] = $order->getClientId();
-        $details[Constants::FIELD_PAID] = true;
+        $details[Constants::FIELD_PAID] = $details['paid'];
```
Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: if you use `->defaults` you do not need this line.

**Comment by antonioperic** [comment]
now ok?

File: src/Payum/Offline/Action/FillOrderDetailsAction.php (hunk 2 above)
Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: @antonioperic ping

**Comment by antonioperic** [comment]
@makasim ping?

**Comment by makasim** [comment]
@antonioperic please answer the question I asked here https://github.com/Payum/Payum/pull/310/files#r24070693

this not clear too me

File: src/Payum/Offline/Action/FillOrderDetailsAction.php (hunk 1 above)
Comment (antonioperic) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: I think that is not case, cause payment is not completed until you receive money do basically paid is false until then

Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: @antonioperic 

`if($request->getToken()->getDetails()) {` - has to be removed, this class should not be aware of token and any such things,

we can set paid to false as default, Just mention it in the  UPGRADE

**Comment by antonioperic** [comment]
@makasim I think we got lost here ... ;);) please see is now ok, I removed that line and set just default.

File: src/Payum/Offline/Action/FillOrderDetailsAction.php [hunk 4]
```diff
@@ -26,7 +26,10 @@ public function execute($request)
         $details['description'] = $order->getDescription();
         $details['client_email'] = $order->getClientEmail();
         $details['client_id'] = $order->getClientId();
-        $details[Constants::FIELD_PAID] = true;
+        
+        $details->defaults([
```
Comment (makasim) [inline src/Payum/Offline/Action/FillOrderDetailsAction.php]: dont use `[]` but `arra(). since payum must be compatible with 5.3. sigh

**Comment by antonioperic** [comment]
ping @makasim

**Comment by antonioperic** [comment]
ping @makasim

**Comment by makasim** [comment]
@antonioperic tests still fail

**Comment by antonioperic** [comment]
have idea why?

**Comment by antonioperic** [comment]
cc @makasim

**Comment by makasim** [comment]
you have to wrap details by ArrayObject. something like this $details = ArrayObject::ensureArrayObject($details);

and after you are done set the array back to Order like this; $order->setDetails((array) $details);

**Comment by antonioperic** [comment]
you mean like this



    public function execute($request)
    {
        RequestNotSupportedException::assertSupports($this, $request);

        $order = $request->getOrder();



        $details = $order->getDetails();
> details = ArrayObject::ensureArrayObject($details);


        $details['amount'] = $order->getTotalAmount();
        $details['currency'] = $order->getCurrencyCode();
        $details['number'] = $order->getNumber();
        $details['description'] = $order->getDescription();
        $details['client_email'] = $order->getClientEmail();
        $details['client_id'] = $order->getClientId();
        
        $details->defaults(array(
            Constants::FIELD_PAID => true,
        ));

        $order->setDetails($details);
    }
-- 
Antonio Perić-Mažar, mag.comp.ing
CEO @ Locastic - we help clients develop amazing web and mobile apps
 
Mob: +385 98 813 684
Skype: antonio.peric
 
Stinice 12/II - 21000 Split - Croatia
Tel: +38521 782 059
www.locastic.com <http://www.locastic.com/>
 
Facebook  <http://facebook.com/Locastic> -  Twitter  <http://twitter.com/Locastic> -  Google+  <http://google.com/+Locastic> -  Linkedin <http://linkedin.com/company/Locastic>

**Comment by makasim** [comment]
:/ I will take care of that a bit later.

**Comment by antonioperic** [comment]
ok, thnx
-- 
Antonio Perić-Mažar, mag.comp.ing
CEO @ Locastic - we help clients develop amazing web and mobile apps
 
Mob: +385 98 813 684
Skype: antonio.peric
 
Stinice 12/II - 21000 Split - Croatia
Tel: +38521 782 059
www.locastic.com <http://www.locastic.com/>
 
Facebook  <http://facebook.com/Locastic> -  Twitter  <http://twitter.com/Locastic> -  Google+  <http://google.com/+Locastic> -  Linkedin <http://linkedin.com/company/Locastic>

**Pull request merged**